flows/field.go

```diff
@@ -76,11 +76,17 @@ func (v *FieldValue) TypedValue() types.XValue {
 			return *v.Number
 		}
 	case assets.FieldTypeState:
-		return v.State
+		if v.State != "" {
+			return v.State
+		}
 	case assets.FieldTypeDistrict:
-		return v.District
+		if v.District != "" {
+			return v.District
+		}
 	case assets.FieldTypeWard:
-		return v.Ward
+		if v.Ward != "" {
+			return v.Ward
+		}
```

Comment:
this method now always returns `nil` if a value has no value in its field's type